python/src/scipp/__init__.py

```diff
@@ -36,5 +36,10 @@
 setattr(Dataset, '_repr_html_', make_html)
 setattr(DatasetConstView, '_repr_html_', make_html)
 
-from .io.hdf5 import data_array_to_hdf5 as _data_array_to_hdf5
-setattr(DataArray, 'to_hdf5', _data_array_to_hdf5)
+from .io.hdf5 import to_hdf5 as _to_hdf5
+setattr(Variable, 'to_hdf5', _to_hdf5)
+setattr(VariableView, 'to_hdf5', _to_hdf5)
+setattr(DataArray, 'to_hdf5', _to_hdf5)
+setattr(DataArrayView, 'to_hdf5', _to_hdf5)
+setattr(Dataset, 'to_hdf5', _to_hdf5)
+setattr(DatasetView, 'to_hdf5', _to_hdf5)
```

Comment:
Originally didn't like that you have attributes for these on scipp objects, however under hood, you have provided good separation of IO from the domain types in the implementation.

Reply:
Yes, was not sure about this either, might remove it again, or rather, we should have a wider discussion on our general approach to this, because we have more, similar cases.